schema-registry/common/src/main/java/com/hortonworks/registries/schemaregistry/AggregatedSchemaBranch.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2017 Hortonworks.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package com.hortonworks.registries.schemaregistry;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import java.io.Serializable;
+import java.util.Collection;
+
+/*
+     This class represents the aggregated information about the schema branch which includes all the schema versions tied
+     to that branch and the schema version from which the branch was created
+ */
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AggregatedSchemaBranch implements Serializable {
+
+    private static final long serialVersionUID = -4013250198049024761L;
+
+    private SchemaBranch schemaBranch;
+    private Long rootSchemaVersion;
+    private Collection<SchemaVersionInfo> schemaVersionInfos;
+
+    public AggregatedSchemaBranch(SchemaBranch schemaBranch, Long rootSchemaVersion, Collection<SchemaVersionInfo> schemaVersionInfos) {
```

Comment:
This should have no-arg default constructor for creating these objects from json text. Same needs to be also fixed in `AggregatedSchemaMetadataInfo`. 

Pl add a test to check that. I tried below to check that and it gets failed. 

```
        String aggregateSchemaMetadataStr = new ObjectMapper().writeValueAsString(aggregatedSchemaMetadataCollection);
        LOG.info("aggregateSchemaMetadataStr = " + aggregateSchemaMetadataStr);

        Collection<AggregatedSchemaMetadataInfo> returnedResult =
                new ObjectMapper().readValue(aggregateSchemaMetadataStr,
                                             new TypeReference<Collection<AggregatedSchemaMetadataInfo>>() {});

        LOG.info("returnedResult = " + returnedResult);
        for (AggregatedSchemaMetadataInfo aggregatedSchemaMetadataInfo : returnedResult) {
            LOG.info("aggregatedSchemaMetadataInfo = " + aggregatedSchemaMetadataInfo);
        }
```